curta/src/chip/trace/writer/mod.rs

```diff
@@ -19,8 +19,75 @@ use crate::trace::window::TraceWindow;
 use crate::trace::window_parser::TraceWindowParser;
 use crate::trace::AirTrace;
 
+pub mod row;
+pub mod window;
+
 pub trait AirWriter<F: Field> {
-    fn write_slice(&self, memory_slice: &MemorySlice, value: &[F]);
+    fn write_slice(&mut self, memory_slice: &MemorySlice, value: &[F]);
```

Comment:
nit:  Should `write_slice` and `read_slice` be renamed to `write_memory` and `read_memory`?

Seems like you do follow that naming convention for `LogEntry` (and I suppose implicitly for all the other `read` and `write` functions where if the "object" is not specified, then its `registers`).